src/lib/distance/__test__/distance.test.ts

```diff
@@ -0,0 +1,100 @@
+import { ConversionError } from '../../../exceptions/ConversionError';
+import { deg2rad, parseElementCoords } from '../base';
+
+describe('Unit tests for the distance library', () => {
+  it('tries to convert degrees to radians with invalid values', () => {
+    const testCases = ['thisIsNotANumber', null, NaN];
+
+    testCases.forEach((value) => {
+      try {
+        deg2rad(value as number | undefined);
+      } catch (error) {
+        expect(error).toBeInstanceOf(ConversionError);
+        expect((error as ConversionError).message).toBe('A valid angle in degrees must be provided!');
+      }
+    });
+  });
+
+  it('tries to parse valid GPS elements', () => {
+    const validElements = [
+      [
+        { lat: 35.6544, longitude: 139.74477 }, // elementA
+        { latitude: 21.4225, lng: 39.8261 }, // elementB
+      ],
+      [
+        { lat: 35.6544, lng: '139.74477' },
+        { lat: '21.4225', lng: 39.8261 },
+      ],
+      [
+        { lat: 35.6544, longitude: 139.74477 },
+        { latitude: 21.4225, lng: '39.8261' },
+      ],
+      [
+        { lat: '35.6544', longitude: '139.74477' },
+        { latitude: 21.4225, longitude: 39.8261 },
+      ],
+    ];
+
+    const expected = {
+      A: {
+        lat: deg2rad(35.6544),
+        lng: deg2rad(139.74477),
+      },
+      B: {
+        lat: deg2rad(21.4225),
+        lng: deg2rad(39.8261),
+      },
+    };
+
+    validElements.forEach(([elmA, elmB]) => {
```

Comment:
utilização do forEach não é muito recomendável devido a menor performance comparado a outros loops
 https://leanylabs.com/blog/js-forEach-map-reduce-vs-for-for_of/

- [ ] Refactor forEach statements